commit 712b3f43ba0e5f86db69cf601ffb06085c606642
Author: Justin Stitt <justinstitt@google.com>
Date:   Mon Oct 30 20:40:48 2023 +0000

    scsi: ibmvscsi: Replace deprecated strncpy() with strscpy()
    
    strncpy() is deprecated for use on NUL-terminated destination strings [1]
    and as such we should prefer more robust and less ambiguous string
    interfaces.
    
    We expect partition_name to be NUL-terminated based on its usage with
    format strings:
    |       dev_info(hostdata->dev, "host srp version: %s, "
    |                "host partition %s (%d), OS %d, max io %u\n",
    |                hostdata->madapter_info.srp_version,
    |                hostdata->madapter_info.partition_name,
    |                be32_to_cpu(hostdata->madapter_info.partition_number),
    |                be32_to_cpu(hostdata->madapter_info.os_type),
    |                be32_to_cpu(hostdata->madapter_info.port_max_txu[0]));
    ...
    |       len = snprintf(buf, PAGE_SIZE, "%s\n",
    |                hostdata->madapter_info.partition_name);
    
    Moreover, NUL-padding is not required as madapter_info is explicitly
    memset to 0:
    |       memset(&hostdata->madapter_info, 0x00,
    |                       sizeof(hostdata->madapter_info));
    
    Considering the above, a suitable replacement is strscpy() [2] due to the
    fact that it guarantees NUL-termination on the destination buffer without
    unnecessarily NUL-padding.
    
    Link: https://www.kernel.org/doc/html/latest/process/deprecated.html#strncpy-on-nul-terminated-strings [1]
    Link: https://manpages.debian.org/testing/linux-manual-4.8/strscpy.9.en.html [2]
    Link: https://github.com/KSPP/linux/issues/90
    Cc: <linux-hardening@vger.kernel.org>
    Signed-off-by: Justin Stitt <justinstitt@google.com>
    Link: https://lore.kernel.org/r/20231030-strncpy-drivers-scsi-ibmvscsi-ibmvscsi-c-v1-1-f8b06ae9e3d5@google.com
    Reviewed-by: Kees Cook <keescook@chromium.org>
    Tested-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>

diff --git a/drivers/scsi/ibmvscsi/ibmvscsi.c b/drivers/scsi/ibmvscsi/ibmvscsi.c
index 59599299615d..71f3e9563520 100644
--- a/drivers/scsi/ibmvscsi/ibmvscsi.c
+++ b/drivers/scsi/ibmvscsi/ibmvscsi.c
@@ -266,7 +266,7 @@ static void set_adapter_info(struct ibmvscsi_host_data *hostdata)
 	dev_info(hostdata->dev, "SRP_VERSION: %s\n", SRP_VERSION);
 	strcpy(hostdata->madapter_info.srp_version, SRP_VERSION);
 
-	strncpy(hostdata->madapter_info.partition_name, partition_name,
+	strscpy(hostdata->madapter_info.partition_name, partition_name,
 			sizeof(hostdata->madapter_info.partition_name));
 
 	hostdata->madapter_info.partition_number =